docs/reference/garbage-collection.md

```diff
@@ -75,6 +71,11 @@ spark-submit --class io.treeverse.clients.GarbageCollector \
   example-repo us-east-1
 ```
 ## Considerations
+1. In order for an object to be hard-deleted, it must be deleted from all branches.
+   You should remove stale branches to prevent them from retaining old objects.
+   For example, consider a branch that has been merged to `main` and has become stale.
+   An object which is later deleted from `main` will always be present in the stale branch, prevent from being hard-deleted.
```

Comment:
This may be painful.  Perhaps lessen the pain by suggesting here how to configure a _default_ retention for all branches?  Then I can set the default to less than the value for `main`, and now "unimportant" data will eventually go away.